Cesium3DTilesSelection/include/Cesium3DTilesSelection/WebMapTileServiceRasterOverlay.h

```diff
@@ -0,0 +1,161 @@
+
+#include <functional>
+#include <memory>
+
+namespace Cesium3DTilesSelection {
+
+class CreditSystem;
+
+/**
+ * @brief Options for tile map service accesses.
+ */
+struct WebMapTileServiceRasterOverlayOptions {
+
+  /**
+   * @brief The MIME type for images to retrieve from the server.
+   */
+  std::optional<std::string> format;
+
+  /**
+   * @brief The subdomains to use for the <code>{s}</code> placeholder in the URL template.
+   *
+   * If this parameter is a single string, each character in the string is a
+   * subdomain. If it is an array, each element in the array is a subdomain.
+   */
+  std::optional<std::variant<std::string, std::vector<std::string>>> subdomains;
+
+  /**
+   * @brief A credit for the data source, which is displayed on the canvas.
+   */
+  std::optional<std::string> credit;
+
+  /**
+   * @brief The layer name for WMTS requests.
+   */
+  std::optional<std::string> layer;
+
+  /**
+   * @brief The style name for WMTS requests.
+   */
+  std::optional<std::string> style;
+
+  /**
+   * @brief The identifier of the TileMatrixSet to use for WMTS requests.
+   */
+  std::optional<std::string> tileMatrixSetID;
+
+  /**
+   * @brief A list of identifiers in the TileMatrix to use for WMTS requests,
+   * one per TileMatrix level.
+   */
+  std::optional<std::vector<std::string>> tileMatrixLabels;
+
+  /**
+   * @brief The minimum level-of-detail supported by the imagery provider.
+   *
+   * Take care when specifying this that the number of tiles at the minimum
+   * level is small, such as four or less. A larger number is likely to
+   * result in rendering problems.
+   */
+  std::optional<uint32_t> minimumLevel;
+
+  /**
+   * @brief The maximum level-of-detail supported by the imagery provider.
+   *
+   * This will be `std::nullopt` if there is no limit.
+   */
+  std::optional<uint32_t> maximumLevel;
+
+  /**
+   * @brief The {@link CesiumGeometry::Rectangle}, in radians, covered by the
+   * image.
+   */
+  std::optional<CesiumGeometry::Rectangle> coverageRectangle;
+
+  /**
+   * @brief The {@link CesiumGeospatial::Projection} that is used.
+   */
+  std::optional<CesiumGeospatial::Projection> projection;
+
+  /**
+   * @brief The {@link CesiumGeometry::QuadtreeTilingScheme} specifying how
+   * the ellipsoidal surface is broken into tiles.
+   */
+  std::optional<CesiumGeometry::QuadtreeTilingScheme> tilingScheme;
+
+  /**
+   * @brief The {@link CesiumGeospatial::Ellipsoid}.
+   *
+   * If the `tilingScheme` is specified, this parameter is ignored and
+   * the tiling scheme's ellipsoid is used instead. If neither parameter
+   * is specified, the {@link CesiumGeospatial::Ellipsoid::WGS84} is used.
+   */
+  std::optional<CesiumGeospatial::Ellipsoid> ellipsoid;
+
+  /**
+   * @brief A object containing static dimensions and their values.
+   */
+  std::optional<std::map<std::string, std::string>> dimensions;
+
+  /**
+   * @brief Pixel width of image tiles.
+   */
+  std::optional<uint32_t> tileWidth;
```

Comment:
For parameters that are not required because they have defaults (like this one), you should mention in the docs what default value is used.